packages/types/src/payment.ts

```diff
@@ -0,0 +1,63 @@
+import { z } from "zod"
+
+export const ProductPaymentProviderSchema = z.object({
+  productId: z.string().uuid(),
+  paymentProvider: z.enum(["STRIPE"]), // include VIPPS later
+  paymentProviderId: z.string(), // Stripe = public_key | Vipps = client_id
+})
+
+export type ProductPaymentProvider = z.infer<typeof ProductPaymentProviderSchema>
+
+export const PaymentProviderSchema = ProductPaymentProviderSchema.omit({
+  productId: true,
+})
+
+export type PaymentProvider = z.infer<typeof PaymentProviderSchema>
+
+export const ProductPaymentProviderWriteSchema = ProductPaymentProviderSchema
+
+export type ProductPaymentProviderWrite = z.infer<typeof ProductPaymentProviderWriteSchema>
+
+export const ProductSchema = z.object({
+  id: z.string().uuid(),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+  type: z.enum(["EVENT"]), // inlude WEBSHOP later
+  objectId: z.string().uuid().nullable(), // The OW object this product is linked to e.g. eventId, webshopItemId. null if not linked
+  amount: z.number(), // price
+  paymentProviders: z.array(PaymentProviderSchema),
+  deletedAt: z.date().nullable(), // null = not deleted
+})
+
+export type Product = z.infer<typeof ProductSchema>
+
+export const ProductWriteSchema = ProductSchema.partial({
```

Comment:
Probably should be omit, as these should never be able to be updateable,

Reply:
Fixed